server/game/GameActions/RevealCards.js

```diff
@@ -0,0 +1,107 @@
+                if(context.revealed.length > 0) {
+                    context.game.queueStep(new AcknowledgeRevealCardsPrompt(context.game, context.revealed, context.player))
+                }
+            }
+        });
+    }
+
+    canChangeGameState({ cards }) {
+        cards = Array.isArray(cards) ? cards : [cards];
+        return cards.length > 0 && cards.some(card => this.isInHiddenArea(card));
+    }
+
+    createEvent({ cards, player, whileRevealed, context }) {
+        const allPlayers = context.game.getPlayers();
+        const eventParams = {
+            player,
+            opponents: allPlayers.filter(p => p !== player),
+            cards: Array.isArray(cards) ? cards : [cards],
+            source: context.source
+        }
+        return this.event('onCardsRevealed', eventParams, event => {
+            const whileRevealedGameAction = whileRevealed || this.defaultWhileRevealed;
+            const playerRevealFunc = (card, player) => event.cards.includes(card) && player === event.player;
+            const allPlayersRevealFunc = card => event.cards.includes(card);
+
+            // Initially make cards visible to the revealing player before the actual "reveal" event. This ensures any interrupts to 'onCardRevealed' will actually show the card (eg. Alla Tyrell)
+            context.game.cardVisibility.addRule(playerRevealFunc);
+            
+            for(let card of event.cards) {
+                const revealEventParams = {
+                    card,
+                    cardStateWhenRevealed: card.createSnapshot(),
+                    player: event.player,
+                    opponents: event.opponents,
+                    source: event.source
+                }
+                
+                event.thenAttachEvent(this.event('onCardRevealed', revealEventParams, () => {}));
+            }
+            context = Object.assign({ revealed: event.cards, player: event.player, opponents: event.opponents }, context);
+
+            // Highlight & reveal cards simultaneously with onCardRevealed (after interrupts)
+            let prepareRevealCards = new HandlerGameActionWrapper({ handler: context => {
+                    // Remove visibility for revealing player
+                    context.game.cardVisibility.removeRule(playerRevealFunc);
+                    // Filter out any cards that are no longer hidden (eg. Alla Tyrell)
+                    context.revealed = context.revealed.filter(card => this.isInHiddenArea(card));
+                    event.revealed = context.revealed;
+                    // Reveal all remaining cards to all players
+                    if(context.revealed.length > 0) {
+                        context.game.cardVisibility.addRule(allPlayersRevealFunc);
+                        this.highlightRevealedCards(event, context.revealed, allPlayers);
+                    }
+                }
+            });
+            event.thenAttachEvent(prepareRevealCards.createEvent(context));
```

Comment:
No need to wrap this in a `HandlerGameActionWrapper`. Since we always want to do this and don't need to check the `allow`, you can just create an event directly.

Reply:
This event has been removed in e2661e091c5a94eb156a653cf08385a597cf4e74, as it's luckily not required anymore due to revealing happening before reveal interrupts, with `context.revealed` being filtered within the  `onCardRevealed` event.